packages/office-ui-fabric-react/src/components/Panel/Panel.types.ts

```diff
@@ -284,3 +297,146 @@ export enum PanelType {
    */
   custom = 7
 }
+
+export interface IPanelStyleProps {
+  /**
+   * Theme provided by High-Order Component.
+   */
+  theme: ITheme;
+
+  /**
+   * Accept custom classNames
+   */
+  className?: string;
+
+  /**
+   * Is Panel open
+   */
+  isOpen?: boolean;
+
+  /**
+   * Is animation currently running
+   */
+  isAnimating?: boolean;
+
+  /**
+   * Is panel on right side
+   */
+  isOnRightSide?: boolean;
+
+  /**
+   * Is panel hidden on dismiss
+   */
+  isHiddenOnDismiss?: boolean;
```

Comment:
Non-blocking but I'd consider using `Pick<IPanelProps>` for the common attributes